diff --git a/facetapi.js b/facetapi.js
index 85d941e..21a498b 100644
--- a/facetapi.js
+++ b/facetapi.js
@@ -42,14 +42,14 @@ Drupal.facetapi.applyLimit = function(settings) {
       return $(this).find('li').length > settings.limit;
     }).each(function() {
       $('<a href="#" class="facetapi-limit-link"></a>').text(Drupal.t(settings.showMoreText)).click(function() {
-        if ($(this).siblings().find('li:hidden').length > 0) {
-          $(this).siblings().find('li:gt(' + limit + ')').slideDown();
-          $(this).addClass('open').text(Drupal.t(settings.showFewerText));
-        }
-        else {
+        if ($(this).hasClass('open')) {
           $(this).siblings().find('li:gt(' + limit + ')').slideUp();
           $(this).removeClass('open').text(Drupal.t(settings.showMoreText));
         }
+        else {
+          $(this).siblings().find('li:gt(' + limit + ')').slideDown();
+          $(this).addClass('open').text(Drupal.t(settings.showFewerText));
+        }
         return false;
       }).insertAfter($(this));
     });
